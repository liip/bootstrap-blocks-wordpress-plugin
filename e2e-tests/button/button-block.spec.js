/**
 * WordPress dependencies
 */
import {
	clickBlockToolbarButton,
	clickButton,
	createNewPost,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';
import { selectOption } from '../helper';
import { insertButtonBlock, selectButtonBlock } from './button-helper';

describe( 'button block', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'Button block should be initialized with default attributes', async () => {
		await insertButtonBlock();
		await selectButtonBlock();

		// Editor content should match snapshot
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Should be possible to set link url and text', async () => {
		await insertButtonBlock();
		await selectButtonBlock();

		// Set button text
		await page.type(
			'[aria-label="Add text..."].block-editor-rich-text__editable',
			'Liip'
		);

		// Editor content should match snapshot
		expect( await getEditedPostContent() ).toMatchSnapshot();

		// Set button url
		await page.type( 'input[aria-label="URL"]', 'https://liip.ch' );

		// Editor content should match snapshot
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Should be possible to select style', async () => {
		await insertButtonBlock();
		await selectButtonBlock();

		// Style option should be applied
		await selectOption( 'Style', 'secondary' );

		// Editor content should match snapshot
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	// TODO fix broken test
	it.skip( 'Should be possible to change alignment', async () => {
		await insertButtonBlock();
		await selectButtonBlock();

		// Change alignment
		await clickBlockToolbarButton( 'Change button alignment' );
		await clickButton( 'Align text center' );
		expect(
			await page.$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/button"][data-alignment="center"]'
			)
		).not.toBeNull();

		// Editor content should match snapshot
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
