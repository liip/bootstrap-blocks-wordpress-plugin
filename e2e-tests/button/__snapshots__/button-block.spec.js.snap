// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button block Button block should be initialized with default attributes 1`] = `"<!-- wp:wp-bootstrap-blocks/button /-->"`;

exports[`button block Should be possible to select style 1`] = `"<!-- wp:wp-bootstrap-blocks/button {\\"style\\":\\"secondary\\"} /-->"`;

exports[`button block Should be possible to set link url and text 1`] = `"<!-- wp:wp-bootstrap-blocks/button {\\"text\\":\\"Liip\\"} /-->"`;

exports[`button block Should be possible to set link url and text 2`] = `"<!-- wp:wp-bootstrap-blocks/button {\\"url\\":\\"https://liip.ch\\",\\"text\\":\\"Liip\\"} /-->"`;
